native-image-config/build.gradle.kts

```diff
@@ -0,0 +1,335 @@
+        add("""^io\.grpc\.okhttp\.""".toRegex())
+        add("""^munit\.""".toRegex())
+        add("""^org\.assertj\.""".toRegex())
+        add("""^org\.mockito\.""".toRegex())
+        add("""^org\.gradle\.""".toRegex())
+        add("""^testing\.""".toRegex())
+        add("""^worker\.org\.gradle\.""".toRegex())
+    }
+
+    excludedResourceRegexes.apply {
+        // Exclude the resource files that are referenced only from the tests.
+        add("""Test(?:Utils?)?\.""".toRegex())
+        add("""^test(?:ing)?[/\\.]""".toRegex())
+        add("""^CatalogManager\.properties$""".toRegex())
+        add("""^META-INF/armeria/grpc$""".toRegex())
+        add("""^META-INF/dgminfo""".toRegex())
+        add("""^META-INF/groovyx?/""".toRegex())
+        add("""^META-INF/native/libio_grpc_netty_""".toRegex())
+        add("""^META-INF/services/net\.javacrumbs\.jsonunit\.""".toRegex())
+        add("""^META-INF/services/org\.apache\.groovy\.""".toRegex())
+        add("""^META-INF/services/org\.apache\.xerces\.""".toRegex())
+        add("""^META-INF/services/org\.assertj\.""".toRegex())
+        add("""^META-INF/services/org\.codehaus\.groovy\.""".toRegex())
+        add("""^META-INF/services/org\.junit(?:pioneer)?\.""".toRegex())
+        add("""^catalog/""".toRegex())
+        add("""^jndi\.properties$""".toRegex())
+        add("""^junit-platform\.properties$""".toRegex())
+        add("""^logback-test\.xml$""".toRegex())
+        add("""^mockito-extensions/""".toRegex())
+        add("""^mozilla/""".toRegex())
+        add("""^org/apache/hc/""".toRegex())
+        add("""^org/apache/http/""".toRegex())
+        add("""^org/apache/xml/""".toRegex())
+    }
+}
+val cleanupNativeImageConfigTask = tasks.named("cleanupNativeImageConfig", CleanupNativeImageConfigTask::class)
+
+tasks.register("nativeImageConfig", Exec::class).configure {
+    group = "Build"
+    description = "Generates the final native image config by merging the base and generated native image config."
+
+    dependsOn(cleanupNativeImageConfigTask)
+    inputs.dir(project.file("src/base-config"))
+    outputs.dir(nativeImageConfigOutputDir)
+
+    commandLine(
+            nativeImageConfigToolPath,
+            "generate",
+            "--input-dir=src/base-config",
+            "--input-dir=$cleanupNativeImageConfigOutputDir",
+            "--output-dir=$nativeImageConfigOutputDir"
+    )
+
+    doFirst {
+        // Delete the output directory because otherwise the tool doesn't overwrite the files.
+        delete(nativeImageConfigOutputDir)
+    }
+}
+
+configure(relocatedProjects) {
+    val relocatedProject = this
+    processNativeImageTracesTask.configure {
+        dependsOn(relocatedProject.tasks["nativeImageTrace"])
+        val nativeImageTraceFile = relocatedProject.ext["nativeImageTraceFile"] as Path
+        nativeImageTraceFiles.add(nativeImageTraceFile)
+    }
+}
+
+abstract class CleanupNativeImageConfigTask : DefaultTask() {
```

Comment:
perhaps `FilterAndCopyNativeImageConfigTask`?
For me, the word `clean` in a Gradle task means removing the generated files. 🤔